### DISCUSSION TITLE
Experience Fragment reference provider behavior

### DISCUSSION THREAD

**Discussion by michaltobiasz** [adobe/aem-core-wcm-components#1399] [Q&A] [answered]
I have some doubts if the reference mechanism used for experience fragments to provide references when publishing the page works as should.

I want to publish a page ex. _/content/test/gb/en/page1_ in edit mode. The page is based on the dynamic template which contains in the structure experience fragments (header and footer) pointing to language-master
_/content/experience-fragments/test/language-master/en/header_
_/content/experience-fragments/test/language-master/en/footer_

There are also XFs live copies created for _en/gb_
_/content/experience-fragments/test/gb/en/header_
_/content/experience-fragments/test/gb/en/footer_

Experience fragments store in the properties, paths to pages from the given market.
Ex.
_/content/experience-fragments/test/language-master/en/header_ has component with property path pointing to _/content/test/language-master/en/page2_
_/content/experience-fragments/test/gb/en/header_ has component with property path pointing to _/content/test/gb/en/page2_

In case when I publish _/content/test/gb/en/page1_ the reference mechanism is analyzing references from _/content/experience-fragments/test/language-master/en/header_  and it children only. I'm wondering if this mechanism shouldn't check _/content/experience-fragments/test/gb/en/header_ which is indead used by the page and don't check the language-master part.

Any idea if it is correct behavior or XF reference provider mechanism is missing something?

**Comment by ky940819** [comment]
Just to clarify for me,
Do you have XFs being referenced withing XFs?
Or is the issue that it's not picking up the correct localized XF variant, and is always just checking the master, which is what is actually referenced in the page (even if it's not what is rendered)?
Or none of the above?

**Reply by michaltobiasz** [reply]
`Do you have XFs being referenced withing XFs?` - No

Yes. The reference provider is not finding the correct localized XF ex. _/content/experience-fragments/test/gb/en/header_ and it always just looking for references defined in the master _/content/experience-fragments/test/language-master/en/header_ and it child nodes. The path to master is found in the template structure from which the page was created.

**Comment by vladbailescu** [comment, marked as answer]
@michaltobiasz, you are right: the reference provider should resolve to the localised experience fragments, not the masters, when publishing a localised page. Since the provider is not part of Core Components, but it's in the product, I filed an improvement ticket (CQ-4317044). Please reach out to customer support if you want to file a request and have it prioritised higher based on the impact on your work!

**Reply by michaltobiasz** [reply]
@vladbailescu thank you for your response!

**Comment by nowugetme** [comment]
XFs being referenced withing XFs  the reference provider should resolve to the localised experience fragments
https://setkorp.com/
